lib/routes/buaa/lib/space/newbook.ts

```diff
@@ -0,0 +1,171 @@
+  - 默认值：无
+  - 注意事项：只有本馆一个可选值。
+- \`locaCode\`：馆藏地代码
+  - 例：
+    - 五层西-中文新书借阅室(A-Z类)：\`02503\`
+  - 默认值：无
+  - 注意事项：必须与 \`libCode\` 同时使用。
+
+示例：
+- \`buaa/lib/space/newbook\` 为所有新书
+- \`buaa/lib/space/newbook/clsNo=U&libCode=00000&locaCode=60001\` 为沙河教2图书馆所有中图分类号为 U（交通运输）的书籍
+`,
+    categories: ['university'],
+
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportRadar: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+};
+
+async function handler(ctx: Context): Promise<Data> {
+    const path = ctx.req.param('path');
+    const i = path.indexOf('/');
+    const params = i === -1 ? '' : path.slice(i + 1);
+    const searchParams = new URLSearchParams(params);
+    const dcpCode = searchParams.get('dcpCode'); // Filter by subject (discipline code)
+    const clsNo = searchParams.get('clsNo'); // Filter by class (Chinese Library Classification)
+    if (dcpCode && clsNo) {
+        throw new Error('dcpCode and clsNo cannot be used at the same time');
+    }
+    searchParams.set('pageSize', '100'); // Max page size. Any larger value will be ignored
+    searchParams.set('page', '1');
+    !dcpCode && !clsNo && searchParams.set('dcpCode', 'nolimit'); // No classification filter
+    const url = `https://space.lib.buaa.edu.cn/meta-local/opac/new/100/${clsNo ? 'byclass' : 'bysubject'}?${searchParams.toString()}`;
+    const { data } = await got(url);
+    const list = (data?.data?.dataList || []) as Book[];
+    const item = await Promise.all(list.map(async (item: Book) => await getItem(item)));
+    const res: Data = {
+        title: '北航图书馆 - 新书速递',
+        item,
+        description: '北京航空航天大学图书馆新书速递',
+        language: 'zh-CN',
+        link: 'https://space.lib.buaa.edu.cn/space/newBook',
+        author: '北京航空航天大学图书馆',
+        allowEmpty: true,
+        image: 'https://lib.buaa.edu.cn/apple-touch-icon.png',
+    };
+    return res;
+}
+
+async function getItem(item: Book): Promise<DataItem> {
+    const info = await getItemInfo(item.isbn);
+    const holdings = JSON.parse(item.holdings) as Holding[];
+    const link = `https://space.lib.buaa.edu.cn/space/searchDetailLocal/${item.bibId}`;
+    const content = art(path.join(__dirname, 'templates/newbook.art'), {
+        item,
+        info,
+        holdings,
+    });
+    return {
+        language: item.language === 'eng' ? 'en' : 'zh-CN',
+        title: item.title,
+        pubDate: item.onSelfDate ? timezone(parseDate(item.onSelfDate), +8) : undefined,
+        description: content,
+        link,
+    };
+}
+
+async function getItemInfo(isbn: string): Promise<Info | null> {
+    const url = `https://space.lib.buaa.edu.cn/meta-local/opac/third_api/douban/${isbn}/info`;
+    return (await cache.tryGet(url, async () => {
+        const response = await got(url);
+        return JSON.parse(response.body).data;
+    })) as Promise<Info | null>;
```

Comment:
Cache the whole returned object https://github.com/DIYgod/RSSHub/blob/d9658f87b36977295e67787513a36d973e29fa4f/lib/routes/buaa/lib/space/newbook.ts#L156-L162 instead of API response only